src/lib.rs

```diff
@@ -1020,6 +1032,15 @@ mod test {
         assert_eq!(expected, actual);
     }
 
+    #[test]
+    fn arbitrary_for_bytes() {
+        let x = [1, 2, 3, 4, 4];
+        let mut buf = Unstructured::new(&x);
+        let expected = &[1, 2, 3, 4];
+        let actual = <&[u8] as Arbitrary>::arbitrary(&mut buf).unwrap();
+        assert_eq!(expected, actual);
+    }
```

Comment:
Does this test still pass with the length prefix stuff? I expect it to pass when using `arbitrary_take_rest` but not with regular `arbitrary`, where I instead I would expect that we read a length (which is read from the end of the data) and then get some prefix of `[1, 2, 3]` depending on how `arbitrary_len` maps `4` into the `0..3` range.

Reply:
Check the latest commit– does that match what's in your head?